### ISSUE TITLE
Question: LN hot wallet to cold transfer / security

### ISSUE THREAD

**Issue by woutersamaey** [btcpayserver/btcpayserver#476]
As I understand it, LN is a hot wallet, while the btc node (with Ledger Nano S) is cold storage (no private keys on the server).

When running BTCPay Server as a merchant, is there a way provided to periodically move funds over from hot to cold storage without negatively impacting the LN channels or connectivity in the network? 

My knowledge is still a bit limited, but as I understand it, you need to keep funds in channels to keep them open, is it not? Emptying them would then close channels?

How does this all fit together to maximize security for the merchant, and how best to go about it?

Should we create some cronjobbed scripts or something like that?

Any general advice on this would be greatly appreciated, and maybe it would be a good idea to put this in the docs as well?

**Comment by Kukks** [comment]
I remember some people were using the zigzag exchange to swap ln BTC to on
chain btc

**Comment by bitcoinshirt** [comment]
Your question is more about Lightning Network itself than about BTCPay.

You can easily close channels and withdraw the founds to your on-chain wallet and then move the to your cold storage. This is not recommended though as you'll reduce the liquidity of the network.

Maybe a better way would be to use an online exchange like [ZigZag](https://zigzag.io/) as Kukks mentioned. 
There's no security in the LN at the moment, it's experimental and should be handled as such. Don't put the funds you can't afford to loose. I think that was more than clear in the UI and in the documentation.

Basically, you would use ZigZag and "sell" your LN for the on-chain. I think they have a 1% fee for that. I think there are few other LN exchanges, but none that comes to mind mind at this moment.

![screen shot 2018-12-26 at 23 15 24](https://user-images.githubusercontent.com/36959754/50458550-314cb100-0964-11e9-8df4-8b727b7e6380.png)

I don't think there's a way to automate this process for now.

**Comment by woutersamaey** [comment]
Ok, so if I understand correctly, BTCPay Server doesn’t offer any tools or features to deal with this type of situation at this point. 

As LN gets more stable and is used for commerce, these features will at some point be added then?

Using a 3rd party tool which charges 1% fee sounds strange to me, as the whole point was to remove parties. 

Do you see merit in a cronjobbed script (or coded scheduled task) that would for example reduce the funds that are in LN back to a minimal level, keeping an equilibrium between liquidity and security?

If so, perhaps this could be a feature request and have a setting in the UI to keep no more than 1 BTC in LN (for example). Every night, the excess funds could be automatically moved to the onchain wallet, much like emptying a cash drawer and taking the cash to the bank.

**Comment by bitcoinshirt** [comment]
@woutersamaey BTCPay doesn't have an automatic way to withdraw funds from your LN to your on-chain wallet. You can do this manually. 
You should probably play around with LN in order to understand it better.

**Comment by pieman64** [comment]
@woutersamaey zigzag has an API so you can probably automate this if you need to. They appear to charge less than 1% and for some that's a modest sum to "protect" their CDA's.

**Comment by bitcoinshirt** [comment]
The issue is now labeled as "prepare for closing" and will be closed it in the next 7 days if there's no further discussion.  If your question/problem was resolved, you can close the issue yourself. You can re-open the issue at any time.

**Comment by woutersamaey** [comment]
Should this question be turned into a feature request so BTCPay can automatically handle this case for merchants? Some merchants aren’t technically savvy, so my first thought would be “yes”?

How else can people be convinced of using LN if there is no easy way to take their funds out easily?

**Comment by bitcoinshirt** [comment]
I still think you need to actually use the LN to better understand how it works. It seems your requesting a feature without fully getting the point of the Lightning Network.

BTCPay does not limit you in any way to pull the funds out of the LN, you are free to close channels when you want to "cash out",  just not automatically, automatic closing of the channels and withdrawing of the funds make no sense to me.

If I was your customer and I paid an on-chain fee to have unlimited amount of instant and cheap transactions and you just closed the channel on me, I'd be pissed.

Solution for what you want is to use an exchange. Maybe when we add #112 an exchange supporting LN like #354 is added you can instantly withdraw to either on-chain or fiat.

**Comment by woutersamaey** [comment]
Thanks @bitcoinshirt. This was the answer I was looking for. Will follow these other threads and learn more. Definitely makes sense now.
